src/main.rs

```diff
@@ -0,0 +1,468 @@
+use macroquad::{color::*, prelude::*};
+
+#[derive(Clone, PartialEq, Debug)]
+enum CellState {
+    Alive,
+    Dead,
+}
+
+#[derive(Clone)]
+struct Board(Vec<Vec<CellState>>);
+
+impl Board {
+    /// Creates a new board from scratch. All the cells start dead by default.
+    /// Output: A game of life board
+    fn new(width: usize, length: usize) -> Self {
+        let row = vec![CellState::Dead; width];
+        let board = vec![row.clone(); length];
+        Board(board)
+    }
+
+    /// Returns the status of a cell, given its coordinates
+    /// Input: x and y coordinates of the cell
+    /// Output: whether the cell is dead or alive
+    fn get_cell_status(&self, x: usize, y: usize) -> CellState {
+        self.0[y][x].clone()
+    }
+
+    /// Swaps a specific position in an already existing board.
+    /// Input: a mutable reference to the board, and the row and column of the cell to update
+    /// NOT the cell udpate function, this one is intended to be used for the user to manually flip the states of cells before the game starts
+    fn swap_cell_state(&mut self, x: usize, y: usize) {
+        if self.0[y][x] == CellState::Alive {
+            self.0[y][x] = CellState::Dead;
+        } else {
+            self.0[y][x] = CellState::Alive;
+        }
+    }
```

Comment:
Swap usually means swapping two different values. For example, you have two mutable variables and you want to swap their values.

What you are doing here is toggling. Toggling means changing a single binary value to its opposite. For example, toggling a dead cell makes it alive, while toggling an alive cell kills it.

```suggestion
    fn toggle_cell_state(&mut self, x: usize, y: usize) {
        if self.0[y][x] == CellState::Alive {
            self.0[y][x] = CellState::Dead;
        } else {
            self.0[y][x] = CellState::Alive;
        }
    }
```